sympy/core/function.py

```diff
@@ -1139,13 +1139,16 @@ def __new__(cls, expr, *variables, **assumptions):
         while i < len(variables) - 1:  # process up to final Integer
             v, count = variables[i: i + 2]
             iwas = i
+            if isinstance(v, (collections.Iterable, Tuple)):
```

Comment:
A less drastic change would be to require Matrix or Array here.

Reply:
Matrix and Array are already supported, this is not needed for them.